src/ansys/speos/core/bsdf.py

```diff
@@ -0,0 +1,789 @@
+        """List of incident angles for reflection and transmission.
+
+        Returns
+        -------
+        list[Union[list[float], None],Union[list[float], None]]
+            Returns a nested list of incidence angels for reflective and transmittive
+            data if not available the value will be None
+        """
+        if self.has_transmission:
+            t_angle = [bxdf.incident_angle for bxdf in self.btdf]
+        else:
+            t_angle = []
+        if self.has_reflection:
+            r_angle = [bxdf.incident_angle for bxdf in self.brdf]
+        else:
+            r_angle = []
+        return [r_angle, t_angle]
+
+
+class InterpolationEnhancement:
+    """Class to facilitate Specular interpolation enhancement.
+
+    Notes
+    -----
+    **Do not instantiate this class yourself**, This is a Helper class,
+    """
+
+    def __init__(self, bsdf: Union[AnisotropicBSDF], index1: float = 1, index2: float = 1):
+        self._bsdf = bsdf
+        self._index1 = index1
+        self._index2 = index2
+
+        self._data, self._data_reflection, self._data_transmission = (
+            self._create_interpolation_data()
+        )
+
+    def _create_interpolation_data(self):
+        indices = anisotropic_bsdf__v1__pb2.RefractiveIndices(
+            refractive_index_1=self.index1, refractive_index_2=self.index2
+        )
+        self._bsdf._stub.GenerateSpecularInterpolationEnhancementData(indices)
+        data = self._bsdf._stub.GetSpecularInterpolationEnhancementData(Empty())
+        data_r, data_t = None, None
+        if self._bsdf.has_transmission:
+            data_t = data.transmission
+        if self._bsdf.has_reflection:
+            data_r = data.reflection
+        return data, data_r, data_t
+
+    @property
+    def index1(self):
+        """Refractive index on reflection side."""
+        return self._index1
+
+    @index1.setter
+    def index1(self, value):
+        self._index1 = value
+        self._data.refractive_index_1 = value
+
+    @property
+    def index2(self):
+        """Refractive index on transmission side."""
+        return self._index1
+
+    @index2.setter
+    def index2(self, value):
+        self._index2 = value
+        self._data.refractive_index_2 = value
+
+    @property
+    def data_reflection(self):
+        """Interpolation data for reflection.
+
+        Notes
+        -----
+        Please don't set directly use define_interpolation_*** methods
+        """
+        data = {}
+        if isinstance(self._bsdf, AnisotropicBSDF) and self._data_reflection is not None:
+            for i, ani_sample in enumerate(self._data_reflection.anisotropic_samples):
+                data[str(self._bsdf.anisotropic_angles[0][i])] = {}
+                for j, incident in enumerate(ani_sample.incidence_samples):
+                    data[str(self._bsdf.anisotropic_angles[0][i])][
+                        str(self._bsdf.incident_angles[0][(i + 1) * (j + 1) - 1])
```

Comment:
should be:
self._bsdf.incident_angle[0][[i*len(self._data_reflection.anisotropic_samples)+j]
?